legal-api/migrations/versions/1ba027b074a5_add_idp_userid_to_users.py

```diff
@@ -0,0 +1,32 @@
+"""add_idp_userid_to_users
+
+Revision ID: 1ba027b074a5
+Revises: e10f724fad6d
+Create Date: 2022-11-22 12:44:50.751121
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '1ba027b074a5'
+down_revision = 'e10f724fad6d'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('users', sa.Column('idp_userid', sa.String()))
```

Comment:
can we update the column def to the following?  length 256 is what auth user table has. 

`op.add_column('users', sa.Column('idp_userid', sa.String(length=256), nullable=True)`

Reply:
I can update the length. And this column will be unique key later.

Reply:
It's possible for it to be null though right?

Reply:
no, it shouldn't be null. It has to have the value from Keycloak.

Reply:
There is user data from COLIN where we don't have that info.  So it needs to be nullable I think.

Reply:
What's information in the 'sub' column now of Colin user? We can copy the value from 'sub' to this column.

Reply:
"sub" info was not available in COLIN db from what I could see.  This is the table where the user info came from.

![image](https://user-images.githubusercontent.com/6685223/203423590-21448616-c0f9-4aeb-a32f-bb411312adb3.png)

Reply:
Do you have example in prod or test database? I need to look at the data first.

Reply:
Here are some examples in LEAR Prod db.

I think in a lot of instances, there was a userid match and that user was just used so there was a sub.  There are 172 user entries in Prod without a sub and I think a lot of those were brought over from COLIN.

```
+----+--------+---------+--------+-----+----+----+---------------------------------+
|id  |username|firstname|lastname|email|sub |iss |creation_date                    |
+----+--------+---------+--------+-----+----+----+---------------------------------+
|2358|SC70011 |         |        |     |null|null|1995-10-27 07:00:00.000000 +00:00|
|2095|PV81928 |         |        |     |null|null|2000-01-04 08:00:00.000000 +00:00|
|2096|PY75075 |         |        |     |null|null|2011-04-08 07:00:00.000000 +00:00|
|2097|PT81928 |         |        |     |null|null|2011-01-24 08:00:00.000000 +00:00|
+----+--------+---------+--------+-----+----+----+---------------------------------+
```

Reply:
@bolyachevets thought?

Reply:
@argush3 we should copy the username to the new column. I checked the existing database only few username got duplicated.

Reply:
ok, sounds good